elastic/kibana
# [Fleet] Fix migration 7.12 to 7.13 migrate settings

## Summary

Resolve https://github.com/elastic/kibana/issues/100050

During the migration for fleet server we create the `.fleet-policies` before creating the fleet settings with fleet server hosts. Because of that in cloud this resulted with a `.fleet-policies` with the `fleet.*` part.

That PR fix it.

## How to test

with a cloud config in `kibana.dev.yml`
```
xpack.cloud.enabled: true
xpack.cloud.id: 'staging:dXMtZWFzdC0xLmF3cy5mb3VuZC5pbyRjZWM2ZjI2MWE3NGJmMjRjZTMzYmI4ODExYjg0Mjk0ZiRjNmMyY2E2ZDA0MjI0OWFmMGNjN2Q3YTllOTYyNTc0Mw=='
xpack.cloud.deployment_url: '/test/deployment-1'
xpack.cloud.base_url: 'http://test.fr'
```

* start kibana 7.12 and es 7.12  (you can use that command to persist data for the migration `yarn es snapshot --license trial --password changeme -E xpack.security.authc.api_key.enabled=true -E path.data=/tmp/es-data`)
* setup agents in kibana 7.12
* Start kibana and ES on that branch 
* Visit the fleet plugin
* Check policies contains the `fleet.*` part. (`GET .fleet-policies/_search` in the devtools)
